mirror-url.go

```diff
@@ -157,13 +160,25 @@ func deltaSourceTarget(sourceURL string, targetURL string, isForce bool, isFake
 				TargetAlias:   targetAlias,
 				TargetContent: targetContent,
 			}
-		} // else if diffMsg.Diff == differInSecond {
-		// Add for bi-directional mirror.
+		} else if diffMsg.Diff == differInSecond && isRemove {
+			if !isForce && !isFake {
+				// Object removal not allowed if force is not set.
+				mirrorURLsCh <- mirrorURLs{
+					Error: errDeleteNotAllowed(diffMsg.SecondURL),
+				}
+				continue
+			}
+			mirrorURLsCh <- mirrorURLs{
+				TargetAlias:   targetAlias,
+				TargetContent: diffMsg.secondContent,
+			}
+		}
```

Comment:
If none of the conditions match, we should still handle it and throw an error or panic. If not, such errors can be hard to catch. Also switch-case may be better here.